Files/Views/Pages/Properties.xaml.cs

```diff
@@ -146,14 +152,104 @@ private void Properties_Unloaded(object sender, RoutedEventArgs e)
             Unloaded -= Properties_Unloaded;
         }
 
-        private async void GetFolderSize(StorageFolder storageFolder)
+        private async void GetFolderSize(StorageFolder storageFolder, CancellationToken token)
         {
-            var folders = storageFolder.CreateFileQuery(CommonFileQuery.OrderByName);
-            var fileSizeTasks = (await folders.GetFilesAsync()).Select(async file => (await file.GetBasicPropertiesAsync()).Size);
-            var sizes = await Task.WhenAll(fileSizeTasks);
-            var folderSize = sizes.Sum(singleSize => (long)singleSize);
-            ItemProperties.ItemSize = ByteSizeLib.ByteSize.FromBytes(folderSize).ToString();
+
+            var fileSizeTask = Task.Run(async () =>
+            {
+                var size = CalculateFolderSize(storageFolder.Path, token);
+                await Dispatcher.RunAsync(CoreDispatcherPriority.Low, () =>
```

Comment:
Since it's deigned not to display size during calculation for now, there's no need to run this on Dispatcher. The size will finally be set at Line 170.
```diff
-                await Dispatcher.RunAsync(CoreDispatcherPriority.Low, () =>
-                {
-                    ItemProperties.ItemSize = ByteSizeLib.ByteSize.FromBytes(size).ToString();
-                });
```